ui/eureka/eureka.js

```diff
@@ -2955,20 +3675,50 @@ class EurekaTracker {
     const name = document.createElement('span');
     name.classList.add('nm-name');
     name.classList.add('text');
-    name.innerText = this.TransByDispLang(nm.label);
+
+    // Short labels only exist for Save-The-Queen content
+    // Changes names' length depending on users options
+    // If no strings are available, the english short ones will be the default ones
+    if (this.zoneInfo.treatNMsAsSkirmishes) {
+      if (this.options.CompleteNamesSTQ)
+        name.innerText = this.TransByDispLang(nm.label);
+      if (!name.innerText)
+        name.innerText = this.TransByDispLang(nm.shortLabel);
+    } else {
+      name.innerText = this.TransByDispLang(nm.label);
+    }
+
     const progress = document.createElement('span');
     progress.innerText = '';
     progress.classList.add('nm-progress');
     progress.classList.add('text');
     const time = document.createElement('span');
     time.classList.add('nm-time');
     time.classList.add('text');
+    const enriched = document.createElement('span');
+    enriched.classList.add('nm-enriched');
+    enriched.classList.add('text');
 
     if (nm.bunny)
       label.classList.add('bunny');
 
+    // Enriched options for Save-The-Queen content
+    // Adds field note drops, name, id & rarity of those
+    if (this.zoneInfo.treatNMsAsSkirmishes && this.options.EnrichedSTQ && nm.dropsFieldNotes) {
+      for (let i = 0; i < Object.keys(fieldNotesList).length; i++) {
+        if (fieldNotesList[i].id === nm.fieldNotes) {
+          enriched.innerText = ('#' + fieldNotesList[i].id + ' : ' + fieldNotesList[i].shortName + ' - ');
+          for (let i2 = 0; i2 !== fieldNotesList[i].rarity; i2++)
+            enriched.innerHTML += (gRarityIcon);
+        }
+      }
+    }
+
     label.appendChild(icon);
     label.appendChild(name);
+    // We don't need an extra empty span if not in Bozja or if user doesn't need it
+    if (this.options.EnrichedSTQ && this.zoneInfo.treatNMsAsSkirmishes)
+      label.appendChild(enriched);
```

Comment:
```suggestion
    label.appendChild(enriched);
```

I think an extra span is not really a cost to worry about.  I think better to keep the code simpler.

Reply:
I wasn't too sure about this so no worries